js/views/userPage/TypeFilter.js

```diff
@@ -0,0 +1,56 @@
+/* categoryFilter.js and typeFilter.js share quite a bit. Ensure that when one is updated,
+the other is also maintained.
+*/
+
+import $ from 'jquery';
+import loadTemplate from '../../utils/loadTemplate';
+import BaseVw from '../baseVw';
+
+export default class extends BaseVw {
+  constructor(options = {}) {
+    const opts = {
+      ...options,
+    };
+
+    opts.initialState = {
+      types: [],
+      selected: 'all',
+      expanded: false,
+      maxInitiallyVisibleTypes: 6,
+      ...(options.initialState || {}),
```

Comment:
I used to do this as well, but I've found the spread operator to be quite smart where if it's not an object, it no-ops, which is what we want, so this could have just been `...options.initialState`.

Reply:
Gotchya. Yeah, I took this from existing code, but I didn't actually realize it works that way - that spread operator is kickass.